### DISCUSSION TITLE
Version 1.4.9-dev.51

### DISCUSSION THREAD

**Discussion by Weichenpfleger** [perpetualKid/FreeTrainSimulator#111] [General]
After updating to version 1.4.9-dev.51 the program crashes immediately!

Greetings

Weichenpfleger

![Screenshot (1)](https://user-images.githubusercontent.com/72570742/180842393-65e99bd5-a07a-482a-8589-a44ba71cf95d.jpg)

**Comment by perpetualKid** [comment]
seems this relates to the toolbox trackviewer, can you please enable logging and try again, and post the logfile? Can't reproduce at the moment, so wondering where it goes wrong.
(and I realize the error message could need some improvement as well....)

**Reply by perpetualKid** [reply]
> seems this relates to the toolbox trackviewer, can you please enable logging and try again, and post the logfile? Can't reproduce at the moment, so wondering where it goes wrong. (and I realize the error message could need some improvement as well....)

I've found the issue, Monogame did release an update which breaks the shader compilation - will provide a new version shortly

**Comment by perpetualKid** [comment]
fixed in 1.4.9-dev.54

**Comment by Weichenpfleger** [comment]
With version 1.4.9-dev.55 OR starts again.
Thanks PerpetualKid!
Unfortunately I can't get outside sounds (ssource.dat)
hear.
The sounds de ttype.dat are played.

Greetings

Weichenpfleger

**Reply by perpetualKid** [reply]
> With version 1.4.9-dev.55 OR starts again. Thanks PerpetualKid! Unfortunately I can't get outside sounds (ssource.dat) hear. The sounds de ttype.dat are played.
> 
> Greetings
> 
> Weichenpfleger

a fix for the outside sound issue has been applied now in 1.4.9-dev.57, please check and report back. 
Also please don't hesitate to report on other issues or bugs, there is a large amount of changes, and due to complexity not all code paths may be tested sufficiently, apologize.

**Comment by Weichenpfleger** [comment]
Unfortunately no sounds of the "ssourse.dat" are played also with the version "1.4.9-dev.57"!

I have also measured the start times:
Vers. 1.4.9-dev.57 = 01:51min. FPS =52
Vers. T1.4-604 = 00:40min. FPS =60
(vertical syncronisation / no shadows)

Greetings

Weichenpfleger

**Comment by perpetualKid** [comment]
is there a route available which uses sounds from ssource.dat, and works in other OR versions? I'm not familiar with sound management in general, but reading and debugging through the original OR code, there is no place where ssource.dat is read, so I'm trying to understand how is that used and what may be missing here. All the world sounds are referenced from world files (*.ws in \World folder).

As for performance, I'll need to do some more investigation, seems some of the recent monogame updates have an impact on performance. It would be great also to see a logfile!
